ktor-shared/ktor-websockets/common/src/io/ktor/websocket/DefaultWebSocketSession.kt

```diff
@@ -125,10 +125,12 @@ internal class DefaultWebSocketSessionImpl(
             error("WebSocket session $this is already started.")
         }
 
-        LOGGER.trace(
-            "Starting default WebSocketSession($this) " +
-                "with negotiated extensions: ${negotiatedExtensions.joinToString()}"
-        )
+        if (LOGGER.isTraceEnabled) {
+            LOGGER.trace(
+                "Starting default WebSocketSession($this) " +
+                    "with negotiated extensions: ${negotiatedExtensions.joinToString()}"
+            )
+        }
```

Comment:
Instead of duplicating the check here, could you please introduce an inline function in `ktor-utils/common/src/io/ktor/util/logging/Logger.kt`?

```kotlin
/**
 * Check `isTraceEnabled` flag before logging to save some memory allocations.
 */
public inline fun Logger.trace(message: () -> String) {
    if (isTraceEnabled) trace(message())
}
```

Then we can start replacing our other trace calls with this to save on memory allocations.